core/services/functions/listener.go

```diff
@@ -325,7 +327,8 @@ func (l *functionsListener) HandleOffchainRequest(ctx context.Context, request *
 		}
 		return err
 	}
-	return l.handleRequest(ctx, requestId, request.SubscriptionId, subscriptionOwner, RequestFlags{}, &request.Data)
+	// subscription owner is set to sender address
+	return l.handleRequest(ctx, requestId, request.SubscriptionId, senderAddr, RequestFlags{}, &request.Data)
```

Comment:
Since there is currently no subscription being used would this make more sense as a zero address instead of the sender address?

Reply:
We still need the address to validate secrets ownership when processing the request.